docs/api/session.md

```diff
@@ -386,6 +386,14 @@ Returns `String` - The user agent for this session.
 
 Returns `Promise<Buffer>` - resolves with blob data.
 
+#### `ses.downloadURL(url)`
+
+* `url` String
+
+Initiates a download of the resource at `url`.
+The API will generate a [DownloadItem](download-item.md) that can be accessed
+with the [will-download](#event-will-download) event.
```

Comment:
Document how this differs from a download initiated from a frame?

Reply:
I am trying to figure it out. Those checks might be Chromium browser specific.

Reply:
the request headers sent by Electron are the same regardless whether the download is associated with a frame or not (no referer)
```
Connection: keep-alive
Sec-Fetch-Site: none
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) divergent-garbage-integrate/1.0.0 Chrome/78.0.3892.0 Electron/8.0.0-nightly.20190823 Safari/537.36
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US
```

Reply:
https://cs.chromium.org/chromium/src/content/browser/download/download_manager_impl.cc?sq=package:chromium&g=0&l=1288-1297
```cpp
  // Check if the renderer is permitted to request the requested URL.
  if (params->render_process_host_id() >= 0 &&
      !DownloadRequestUtils::IsURLSafe(params->render_process_host_id(),
                                       params->url())) {
    CreateInterruptedDownload(
        std::move(params),
        download::DOWNLOAD_INTERRUPT_REASON_NETWORK_INVALID_REQUEST,
        weak_factory_.GetWeakPtr());
    return;
  }
```